From a96297d26c81b65e156c2dcbc8c55362c66c1262 Mon Sep 17 00:00:00 2001
From: Lukas0610 <mail@lukasberger.at>
Date: Mon, 15 Jan 2018 08:56:15 +0100
Subject: [PATCH 3/4] fs_config: fix fileperms for su-binary

  - Remove the setuid-bit, as cm-14.1 states that the daemonized variant doesn't required it

  - Added OTH/RX-bits to allow non-root/shell-users (meaning apps) to also execute su

Change-Id: I0b2b3a6d835694a9b4605fd455d6fe1a69981cf9
---
 libcutils/fs_config.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libcutils/fs_config.cpp b/libcutils/fs_config.cpp
index cc96ff8..cd0fcf0 100644
--- a/libcutils/fs_config.cpp
+++ b/libcutils/fs_config.cpp
@@ -166,7 +166,7 @@ static const struct fs_path_config android_files[] = {
     // the following two files are INTENTIONALLY set-uid, but they
     // are NOT included on user builds.
     { 06755, AID_ROOT,      AID_ROOT,      0, "system/xbin/procmem" },
-    { 04750, AID_ROOT,      AID_SHELL,     0, "system/xbin/su" },
+    { 00755, AID_ROOT,      AID_SHELL,     0, "system/xbin/su" },
 
     // the following files have enhanced capabilities and ARE included
     // in user builds.
-- 
2.7.4

